examples/large_models/inferentia2/util/inf2_save_split_checkpoints.py

```diff
@@ -62,3 +69,26 @@ def opt_amp_callback(model: OPTForCausalLM, dtype: torch.dtype) -> None:
 tokenizer.save_pretrained(args.save_path)
 
 print(f"Files for '{args.model_name}' have been downloaded to '{args.save_path}'.")
+
+if args.subparser == "generate_neuron_cache":
+    os.environ["NEURONX_CACHE"] = "on"
+    os.environ["NEURONX_DUMP_TO"] = create_directory_if_not_exists(
+        args.neuron_cache_dir
+    )
```

Comment:
According to Mike's update, NEURON_COMPILE_CACHE_URL is official setting. check https://awsdocs-neuron.readthedocs-hosted.com/en/latest/general/arch/neuron-features/neuron-caching.html?highlight=NEURON_COMPILE_CACHE_URL#neuron-persistent-cache

Reply:
I tested this and it seems to work with the latest SDK versions `2.14.*` and not with prior versions `2.12.*` and lower. Since this example is based on Neuron SDK version `2.12`, I believe we can retain `NEURONX_DUMP_TO`.

Reply:
Also tested loading `llama2` using `LlamaForSampling` with Neuron SDK `2.14` and it fails with the following error:
```
RuntimeError: Failed compilation with ['neuronx-cc', '--target=trn1', 'compile', '--framework', 'XLA', '/tmp/neuroncc_compile_workdir/cc7c41fb-c8bf-4adf-b91f-6bb8bf5d7c04/model.MODULE_14cd248f5b2ed6a11af6+56a4bda8.hlo.pb', '--output', '/tmp/neuroncc_compile_workdir/cc7c41fb-c8bf-4adf-b91f-6bb8bf5d7c04/model.MODULE_14cd248f5b2ed6a11af6+56a4bda8.neff', '--model-type=transformer-inference', '--model-type=transformer', '--verbose=35']: 2023-10-25T17:36:47Z Too many instructions after unroll for function sg0000 !
```

Reply:
this root cause of this error is there is bug with 2.14 compiler. "NEURON_COMPILE_CACHE_URL" is still recommended by inf2 team since the old way generates too much debug log. Please keep this PR open once the bug fixing is verified on neuron sdk 2.15 compile.